src/sinks/vector/sink.rs

```diff
@@ -62,7 +63,7 @@ where
                 let builder = RequestMetadataBuilder::new(
                     event_collection.events.len(),
                     event_collection.events_byte_size,
-                    event_collection.events_byte_size, // this is fine as it isn't being used
+                    JsonSize::new(event_collection.events_byte_size), // this is fine as it isn't being used
```

Comment:
Does this comment still apply? I'd expect we'd want/need it to be used at this point.

Reply:
@StephenWakely bump!

Reply:
I'm pretty sure it does. I will revisit shortly.

Reply:
Turns out this was not fine.. New PR coming up.